module/PowerShellEditorServices/Commands/Public/Start-DebugAttachSession.ps1

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+using namespace System.Collections
+using namespace System.Management.Automation
+using namespace System.Reflection
+using namespace System.Threading
+using namespace System.Threading.Tasks
+
+function Start-DebugAttachSession {
+    <#
+    .EXTERNALHELP ..\PowerShellEditorServices.Commands-help.xml
+    #>
+    [OutputType([System.Management.Automation.Job2])]
+    [CmdletBinding(DefaultParameterSetName = 'ProcessId')]
+    param(
+        [Parameter()]
+        [string]
+        $Name,
+
+        [Parameter(ParameterSetName = 'ProcessId')]
+        [int]
+        $ProcessId,
+
+        [Parameter(ParameterSetName = 'CustomPipeName')]
+        [string]
+        $CustomPipeName,
+
+        [Parameter()]
+        [string]
+        $RunspaceName,
+
+        [Parameter()]
+        [int]
+        $RunspaceId,
+
+        [Parameter()]
+        [string]
+        $ComputerName,
+
+        [Parameter()]
+        [switch]
+        $AsJob
+    )
+
+    $ErrorActionPreference = 'Stop'
+
+    try {
+        if ($PSBoundParameters.ContainsKey('RunspaceId') -and $RunspaceName) {
+            $err = [ErrorRecord]::new(
+                [ArgumentException]::new("Cannot specify both RunspaceId and RunspaceName parameters"),
+                "InvalidRunspaceParameters",
+                [ErrorCategory]::InvalidArgument,
+                $null)
+            $err.ErrorDetails = [ErrorDetails]::new("")
+            $err.ErrorDetails.RecommendedAction = 'Specify only one of RunspaceId or RunspaceName.'
```

Comment:
*chefs kiss on recommendedaction*

Reply:
Gotta play to the crowd :) It has made me notice that it's awkward to set it in PowerShell outside of `Write-Error`. Might be something to look into some more.